native/src/components/PinchableBox.tsx

```diff
@@ -0,0 +1,323 @@
+import React, { ReactElement } from 'react'
+import { Animated, Image, ImageErrorEventData, NativeSyntheticEvent } from 'react-native'
+import {
+  PanGestureHandler,
+  PinchGestureHandler,
+  PanGestureHandlerGestureEvent,
+  PanGestureHandlerStateChangeEvent,
+  PinchGestureHandlerGestureEvent,
+  PinchGestureHandlerStateChangeEvent,
+  State
+} from 'react-native-gesture-handler'
+import styled from 'styled-components/native'
+
+const USE_NATIVE_DRIVER = true
+const ANIMATION_DURATION = 150
+
+const Container = styled.View`
+  flex: 1;
+`
+
+type PropsType = {
+  uri: string
+  /**
+   * @param error Error returned by react native of type any
+   */
+  onError: (error: unknown) => void
+}
+
+type StateType = {
+  interactive: boolean
+  viewDimensions?: { width: number; height: number }
+  imageDimensions?: { width: number; height: number }
+}
+
+class PinchableBox extends React.Component<PropsType, StateType> {
```

Comment:
Not sure if `Image` or similar name wouldn't be a better choice, I personally would not look for `PinchableBox` if I'd search for a component to render an image